internal/service/saveURL.go

```diff
@@ -0,0 +1,29 @@
+package service
+
+import (
+	"errors"
+	"github.com/kamencov/go-musthave-shortener-tpl/internal/utils"
+	"log"
+)
+
+func (s *Service) SaveURL(url string) (string, error) {
+	encodeURL, err := utils.EncodeURL(url)
+
+	if err != nil {
+		log.Println(err)
+		return "", errors.New("URL is empty")
```

Comment:
В общем-то не обязательно логировать ошибку непосредственно там, где она возникла. Можно добавить к ошибке контекст и просто вернуть ее из метода, а залогировать в хендлере. К тому же не стоит терять оригинальную ошибку при возврате.

```suggestion
		return "", fmt.Errorf("URL is empty: %v", err)
```